### DISCUSSION TITLE
Missing CreatureSpawnEvent.SpawnReason when using a bucket with tropical fish or axolotl

### DISCUSSION THREAD

**Discussion by marcbal** [PaperMC/Paper#10549] [Ideas]
### Is your feature request related to a problem?

When a player uses a bucket with a fish or axolotl, the event `CreatureSpawnEvent` is fired with the reason `SpawnReason.SPAWNER_EGG`, that is not the real reason.
When we use a plugin that logs specific entity spawns (SPAWNER_EGG for instance), we thought that a player where using spawner-eggs (that is not obtainable in survival) but we figured out that the player was only using buckets in water.

### Describe the solution you'd like.

Add a new entry in the `CreatureSpawnEvent.SpawnReason` enum, maybe like `EMPTY_BUCKET`, `BUCKET` or something named more appropriately.
And of course, calling the event with the appropriate new `SpawnReason` when a player empties a fish/axolotl bucket.

### Describe alternatives you've considered.

Since the player spawning the entity is not referenced by the `CreatureSpawnEvent`, it is not possible to get what’s in the hand of the player, thus it is not possible to deduce if it was a spawner-egg or a bucket.

**Comment by Doc94** [comment]
This looks like a "default" reason, then can make a PR for add a new enum for this cases like SPAWNER_BUCKET (for make the same syntaxis than SPAWNER_EGG)

**Comment by Machine-Maker** [comment]
I think the spawn reason should be preserved through picking up and placing down a mob with a bucket. Now it just resets it to SPAWNER_EGG. If you have it persist the spawn reason in item, I think SPAWNER_EGG is still sensible default, if one isn't present on the stack already.

**Comment by aurorasmiles** [comment]
See also https://github.com/PaperMC/Paper/issues/4361#issuecomment-701640428 regarding this, seems like a similar issue

**Comment by marcbal** [comment]
Thanks for the reference.
I understand that the spawn reason inside the nbt keeps its original value. But for the event itself, I'm expecting to receive the spawn reason associated with the event (and it's still possible to get the "original" spawn reason via Entity#getEntitySpawnReason). I'm confident that this is the expected behaviour in general, since the javadoc of the event doesn't specify it.
